x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -146,6 +156,174 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
+                          },
+                          {
+                            terms: {
+                              [ALERT_INSTANCE_ID]: alerts.map(
+                                (alert) => alert._source['kibana.alert.instance.id']
+                              ),
+                            },
+                          },
+                        ],
+                      },
+                    },
+                    collapse: {
+                      field: ALERT_INSTANCE_ID,
+                    },
+                    sort: [
+                      {
+                        '@timestamp': {
+                          order: 'desc' as const,
+                        },
+                      },
+                    ],
+                  },
+                };
+
+                // We use AlertWithSuppressionFields870 explicitly here as the type instead of
+                // AlertWithSuppressionFieldsLatest since we're reading alerts rather than writing,
+                // so future versions of Kibana may read 8.7.0 version alerts and need to update them
+                const response = await ruleDataClient
+                  .getReader({ namespace: options.spaceId })
+                  .search<typeof suppressionAlertSearchRequest, AlertWithSuppressionFields870<{}>>(
+                    suppressionAlertSearchRequest
+                  );
+
+                const existingAlertsByInstanceId = response.hits.hits.reduce<
+                  Record<string, estypes.SearchHit<AlertWithSuppressionFields870<{}>>>
+                >((acc, hit) => {
+                  acc[hit._source['kibana.alert.instance.id']] = hit;
+                  return acc;
+                }, {});
+
+                const [duplicateAlerts, newAlerts] = partition(
+                  alerts,
+                  (alert) =>
+                    existingAlertsByInstanceId[alert._source['kibana.alert.instance.id']] != null
+                );
+
+                const duplicateAlertUpdates = duplicateAlerts.flatMap((alert) => {
+                  const existingAlert =
+                    existingAlertsByInstanceId[alert._source['kibana.alert.instance.id']];
+                  const existingDocsCount =
+                    existingAlert._source?.[ALERT_SUPPRESSION_DOCS_COUNT] ?? 0;
+                  return [
+                    {
+                      update: {
+                        _id: existingAlert._id,
+                        _index: existingAlert._index,
+                        require_alias: false,
+                      },
+                    },
+                    {
+                      doc: {
+                        [ALERT_LAST_DETECTED]: currentTimeOverride ?? new Date(),
+                        [ALERT_SUPPRESSION_END]: alert._source[ALERT_SUPPRESSION_END],
+                        [ALERT_SUPPRESSION_DOCS_COUNT]:
+                          existingDocsCount + alert._source[ALERT_SUPPRESSION_DOCS_COUNT] + 1,
+                      },
```

Comment:
In the design phase was there any discussion around what to do with the Severity and Risk Score overrides? I.e. should the existing alert's severity/risk score be updated if one of the suppressed alerts matches an increased severity or risk score value?

Reply:
Yeah, we ended up deciding that the suppressed alert doc would be the alert that should be generated from the first source document in the group plus additional metadata to represent the summary stats for the rest of the group. So the severity and risk score are the same values they would be for the first alert in the group if suppression was disabled. For logic that depends on risk score and severity values, e.g. UEBA, the suppression metadata like docs count is available to boost or alter the subsequent calculations in conjunction with the risk score/severity - but the risk score and severity of the actual alert doc are not altered.

Reply:
Perfect -- thanks for the context! 🙂